### ISSUE TITLE
setActiveTab won't add the "pressed" class to the footer button on first click

### ISSUE THREAD

**Issue by landabaso** [01org/appframework#837]
EDIT: This issue is related only with AJAX calls.

Take a look to the code in setActiveTab:

    hash="#"+hash;
    view.find("footer").find("a").removeClass("pressed").attr("data-ignore-pressed", "true")
    .filter([href='"+hash+"']").addClass("pressed");

On first load of an Ajax Call, the href of the footer button will be = to the variable "hash" but without any hash character (#) at the beginning. Thus, the panel will correctly load with the results of an Ajax Call but the "pressed" class won't be added. The code tries to find an href having a # at the beginning (which is not the case on Ajax calls).

EDIT 2:
See an example with the problem here:

http://landabaso.github.io/AFTest
(you may have cached the previous version of the Site. SHIFT+reload for a clean reload).

Click on "AJAX CALL" button to see how it is not activated on the first click.

A possible fix could be to call addClass twice (for hrefs' with or without hashes). Some sanity checks would have to be done first (to find out whether the hrefs' exist or not).

In fact my problem is not related with Ajax calls but it was easier for me to show the problem with that particular case. I'm running AF with my own router making use of pushState only (no hashes, no hashbangs, ...). Only using "/" for navigation without reloads. Everything works pretty OK except for that minor issue in the code.

**Comment by imaffett** [comment]
Thanks for the demo's.  I'll look into this today!  Also, any chance you could post your router somewhere to see ;)  Always excited to see what the community does.

**Comment by landabaso** [comment]
Thank you Ian!

When I said "my own router" I meant one router provided by me instead of AF's router. I'm using this router:
http://krasimirtsonev.com/blog/article/A-modern-JavaScript-router-in-100-lines-history-api-pushState-hash-url
It's a nice router. Only 100 lines of code for all functionality, it doesn't rely on other libs and I found it very easy to integrate with AF.

I simply load the script src for the router above, and then:
```javascript
    $.afui.loadDefaultHash=false;
    $.afui.useInternalRouting=false;
    $.afui.manageHistory=false;
```
And then I set up the router providing routes for my panels where callback functions call loadDiv:
```javascript
Router
.config({ mode: 'history'})
.add(/panelid/, function() {
    $.afui.loadDiv('#panelid', /*new view ?*/ false, /*is back click ?*/ false,
     /*animation*/ "up", '#panelid' /*I believe it is used internally only to find out which view the target belongs to*/)
.listen();
```
In fact I am still not sure about which options should be used in loadDiv (back, newview, and last parameter.

And also you have to capture the "A" clicks and use the router to change state:
```javascript
    $(document).on("click", "a", function(e) {
        e.preventDefault();
        Router.navigate(e.target.getAttribute("href"));
    });
```
Then set the footer hrefs so that they point to "panelid" (not "#panelid) (this conflicts a bit with your feature to auto-load ajax).

And finally you need to setup some .htaccess rules (depending on your server) so that all possible routes point to the same src.

Now I'm working in allowing SEO for AF.
First step is setting this in the header:
```javascript
<meta name="fragment" content="!">
```
so that Google knows this is an ajaxy app.
(more info here: https://developers.google.com/webmasters/ajax-crawling/docs/getting-started)

And then I'll need to run some sort of server which provides the escaped_fragments which google will try to retrieve.

**Comment by imaffett** [comment]
I was thinking about this more this morning.  I think what I'm gonna do is modify it so if it does NOT find a bottom tab with the href='hash' then it will kick out and do nothing.  This will solve the case since we already add the pressed class.

**Comment by landabaso** [comment]
Thanks for the fix Ian. Just a quick note for some other users that may want to use a pushState router in the future with AF3:

Fix #837 almost clears all issues regarding footer buttons not being shown as pressed. Except for one case. If your app starts somewhere such as app.com/A then the footer button for "A" won't be shown as pressed in the first instance. Further clicks will work OK.

You have 2 options.
* Remove the line hash="#"+hash; in setActiveTab
* What I'm doing now - Call this under document.ready:
```javascript
var currentRoute = //get the current route.   
$('#mainView').find("footer").find("a").filter("[href='" + currentRoute + "']").addClass("pressed");
```

Perhaps I'm pushing too much AF3 so that I can use it with a pushState router but I prefer not to fork the project so that I can keep the same codebase as much as possible. Thanks again Ian.

**Comment by imaffett** [comment]
@landabaso  - I said before in another thread, make a change you think fixes it (all around) and submit a pull request.  The worst thing that happens is I accept it ;)
